### DISCUSSION TITLE
CropOrPad with 0 in one of the dims

### DISCUSSION THREAD

**Discussion by jmlipman** [TorchIO-project/torchio#608] [Ideas]
Is there any scenario in which setting 0 to one of the dimensions of CropOrPad makes sense? For example, if you apply CropOrPad((256, 256, 0)) you will get a (256, 256, 0) image, which seems incorrect, isn't it?

I'm planning to modify torchio so that users can indicate with a 0 the dimension in which they don't want to CropOrPad (I initially thought of using -1, but it's a bit more complex because of the function that checks the validity of the provided bounds). In other words, this would be another workaround to use torchio with 2D images, where we might be interested in applying CropOrPad on the slices but not across the slices.

Regards.

**Comment by fepegar** [comment]
Hi, @jmlipman. I think a size of 0 is meaningless, so I agree it could be detected by the transform and do nothing along that direction. Are you interested in contributing with a PR?

Another option is just using `Pad` or `Crop`, if your images have the same 2D size. You can specify `0` for the 3rd dimension. Does that make sense?

**Reply by jmlipman** [reply]
> Another option is just using Pad or Crop, if your images have the same 2D size. You can specify 0 for the 3rd dimension. Does that make sense?

I haven't tried this, but I think it would be great if it can be done already in CropOrPad directly. I have done it already and I think I only needed to add 2 lines to update the target_shape based on source_shape if a dim is 0. I can send a pull request after we are done with the Queue thing : )

**Reply by ashkanpakzad** [reply]
Hi @jmlipman @fepegar  Did this ever get implemented in the end? I'm interested in implementing something similar while calling `CropOrPad` with `mask_name` set to a previously generated lung mask. In my case I'm interested in segmenting structures inside the lung. chest CTs can have several slices above and below that I do not care for. I'm training by extracting 2D axial patches (512,512) on lung chest CT scans where the volumes are typically of size (512,512,x). So I'd be interested in implementing something like 

```tio.CropOrPad((512,512,0), mask_name='lung'),```

to achieve a bounding box crop and reduce the volume size before applying computationally demanding augmentations and then use `UniformSampler` to get the final 2D slices.

Or perhaps this is already achievable some other way?
